sisl/geom/surfaces.py

```diff
@@ -0,0 +1,587 @@
+                            lambda l: 0 if l == ' ' else 1)
+    ]
+
+    iv = 0
+    if geoms_none[0] is None:
+        # our starting geometry will be the 2nd entry
+        geoms_none.pop(0)
+        vacuum = SuperCell([0, 0, vacuums[iv]])
+        out = geoms_none.pop(0).add(vacuum, offset=(0, 0, vacuum.cell[2, 2]))
+        out.set_nsc(c=1)
+        iv += 1
+    else:
+        out = geoms_none.pop(0)
+
+    for geom in geoms_none:
+        if geom is None:
+            dx = out.cell[2, 2] - out.xyz[:, 2].max()
+            # this ensures the vacuum is exactly vacuums[iv]
+            vacuum = SuperCell([0, 0, vacuums[iv] - dx])
+            iv += 1
+            out = out.add(vacuum)
+        else:
+            out = out.append(geom, 2)
+
+    if geoms_none[-1] is None:
+        # ensure that nsc is 1 if there has been inserted vacuum
+        out.set_nsc(c=1)
+
+    return out
+
+
+@set_module("sisl.geom")
+def fcc_slab(alat, atoms, miller, layers=None, vacuum=20., *, orthogonal=False, start=None, end=None):
+    r""" Construction of a surface slab from a face-centered cubic (FCC) crystal
+
+    The slab layers are stacked along the :math:`z`-axis. The default stacking is the first
+    layer as an A-layer, defined as the plane containing an atom at :math:`(x,y)=(0,0)`.
+
+    Parameters
+    ----------
+    alat : float
+        lattice constant of the fcc crystal
+    atoms : Atom
+        the atom that the crystal consists of
+    miller : int or str or (3,)
+        Miller indices of the surface facet
+    layers : int or str, optional
+        Number of layers in the slab or explicit layer specification.
+        An empty character `' '` will be denoted as a vacuum slot, see examples.
+        Currently the layers cannot have stacking faults.
+    vacuum : float or array_like, optional
+        distance added to the third lattice vector to separate
+        the slab from its periodic images. If this is None, the slab will be a fully
+        periodic geometry but with the slab layers. Useful for appending geometries together.
+        If an array layers should be a str, it should be no longer than the number of spaces
+        in `layers`. If shorter the last item will be repeated (like `zip_longest`).
+    orthogonal : bool, optional
+        if True returns an orthogonal lattice
+    start : int or string, optional
+        sets the first layer in the slab. Only one of `start` or `end` must be specified.
+        If set together with `layers` being a str, then they *must* be conforming.
+    end : int or string, optional
+        sets the last layer in the slab. Only one of `start` or `end` must be specified.
+        If set together with `layers` being a str, then they *must* be conforming.
+
+    Examples
+    --------
+    111 surface, starting with the A layer
+    >>> fcc_slab(alat, atoms, "111", start=0)
+
+    111 surface, starting with the B layer
+    >>> fcc_slab(alat, atoms, "111", start=1)
```

Comment:
As a remark, I found that while this line is fine
```python
sisl.geom.fcc_slab(4.08, 79, 111, layers="ABC ABC", start="A")
```
the following lines fail:
```python
sisl.geom.fcc_slab(4.08, 79, 111, layers="AB CABC", start="A")
sisl.geom.fcc_slab(4.08, 79, 111, layers="AB CABC", start=("A", "C"))
```
Is this intuitive?

Reply:
I think this is correct! `start` should be omitted when explicitly defining the `layers`, but it will accept it if `layers[0] == start`, bottom line, if using `layers` as a str, don't supply `start`.

Reply:
I think this is resolved, otherwise, please open again.